amp-pwa-reader/gulpfile.js

```diff
@@ -107,24 +128,28 @@ function injectManifest() {
   });
 }
 
-function watch() {
+function modularizeJS(name) {
+  return "\nmodule.exports = " + name + ';';
+}
 
+function watch() {
   browserSync.init({
-    server: {
-      baseDir: 'dist/',
-      middleware: [historyApiFallback()]
-    },
+    server: false,
     ui: false
   });
 
+  const express = gls.new(paths.server.dest + '/server.js', process.env.PORT || 8080);
+  express.start();
+
   gulp.watch(paths.scripts.src, gulp.series(scripts, inline));
   gulp.watch(paths.styles.src, gulp.series(styles, inline, injectManifest));
   gulp.watch(paths.page.src, dist);
-  gulp.watch(paths.images.src, copy);
-
+  gulp.watch(paths.images.src, dist);   // of course, this could be a smaller task if builds ever got too slow
```

Comment:
I ran into a bug here in the build process... which, I'd like to say, is one of very very few bugs I've run across in this project!

With the old gulpfile, let's say you change an image.  Then the `copy` task runs.  Then the files in the `src` directory get copied into the `dist` directory.  And that's it.  The result is an index.html file with no CSS or JS, since the `inline` task has never run.

This is indeed overkill, but I figured it was good enough for now.  Some folks want to overhaul our gulpfile anyway to make it rely on tasks, not functions, like this:

https://dev-partners.git.corp.google.com/omtinez/stamp-playground/+/master/template/gulpfile.js

I've been planning to create an issue for that.

Reply:
oh good catch! Nevermind then. On tasks vs functions: I actually went from tasks *to* functions :) It's the new hotness in the latest version of Gulp, where tasks are only needed if you need to call them from the command line.